kroxylicious-sample/src/test/java/io/kroxylicious/sample/SampleFilterIntegrationTest.java

```diff
@@ -0,0 +1,154 @@
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.kroxylicious.proxy.config.FilterDefinitionBuilder;
+import io.kroxylicious.test.tester.KroxyliciousTester;
+import io.kroxylicious.testing.kafka.api.KafkaCluster;
+import io.kroxylicious.testing.kafka.junit5ext.KafkaClusterExtension;
+
+import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.proxy;
+import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.withDefaultFilters;
+import static io.kroxylicious.test.tester.KroxyliciousTesters.kroxyliciousTester;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@ExtendWith(KafkaClusterExtension.class)
+public class SampleFilterIntegrationTest {
+
+    private static final String PRE_TRANSFORM_VALUE = "foo";
+    private static final String NO_TRANSFORM_VALUE = "sample";
+    private static final String PRODUCE_TRANSFORM_VALUE = "bar";
+    private static final String FETCH_TRANSFORM_VALUE = "baz";
+    private static final String FIND_CONFIG_FIELD = "findValue";
+    private static final String REPLACE_CONFIG_FIELD = "replaceValue";
+    private static final Map<String, Object> PRODUCE_CONFIG = Map.of(FIND_CONFIG_FIELD, "foo", REPLACE_CONFIG_FIELD, "bar");
+    private static final Map<String, Object> FETCH_CONFIG = Map.of(FIND_CONFIG_FIELD, "bar", REPLACE_CONFIG_FIELD, "baz");
+    private static final Integer TIMEOUT_SECONDS = 10;
+    private static final String TOPIC_NAME = "test";
+    private static final Integer TOPIC_PARTITIONS = 1;
+    private static final Short TOPIC_REPLICATION = 1;
+
+    KafkaCluster cluster;
+    private KroxyliciousTester tester;
+
+    @BeforeEach
+    public void beforeEach() {
+        tester = kroxyliciousTester(withDefaultFilters(proxy(cluster))
+                .addToFilters(new FilterDefinitionBuilder(SampleContributor.SAMPLE_PRODUCE).withConfig(PRODUCE_CONFIG).build())
+                .addToFilters(new FilterDefinitionBuilder(SampleContributor.SAMPLE_FETCH).withConfig(FETCH_CONFIG).build()));
+    }
+
+    @AfterEach
+    public void afterEach() {
+        tester.close();
+    }
+
+    @Test
+    public void sampleFilterWillTransformRoundTripTest(Admin admin) {
+        try (Producer<String, String> producer = tester.producer();
+                Consumer<String, byte[]> kafkaClusterConsumer = new KafkaConsumer<>(mergeMaps(cluster.getKafkaClientConfiguration(),
```

Comment:
Can this consumer be injected by the extension?

Reply:
Looks like it supports Consumer and Producer parameter injection :+1:

Reply:
Yeap, you have an example [here](https://github.com/kroxylicious/kroxylicious/blob/3b1fcfe772784b4c385620d294fb010ec4e7f182/integrationtests/src/test/java/io/kroxylicious/proxy/KroxyStandaloneIT.java#LL48C4-L48C4)

Reply:
There's a bit of a gotcha with this, as I found out yesterday: if you inject the Producer/Consumer, they seem to connect direct to the cluster (i.e. not via Kroxylicious or the filters). I've instead moved the Producer and Consumer setup to `beforeEach()` so they can be spawned from the tester, which doesn't have this problem.

Reply:
Yeah the injected clients will go direct to the backing cluster.

The clients from the tester will go to the Kroxylicious instance